test/runtests.jl

```diff
@@ -35,6 +35,7 @@ import TextParse: getlineend
     @test getlineend("x\nx") == 1
     @test getlineend("x\ny", 2) == 1
     @test getlineend("x\nyz", 3) == 4
+    @test getlineend("xβ\nyz") == 2
```

Comment:
I see you saw this.

The point was that you do something like

```
s = "xβ\nyz"
ii = getlineend("xβ\nyz")
s[1:ii]
```
would get the first line. But with this patch that's no longer the case.

Reply:
So your code example doesn't work with the ``master`` version of the code, and does work with this PR here. ``ii`` has to be a byte index that is a valid character beginning for ``s[1:ii]`` to work. If ``ii`` is the second byte of a multi codeunit character, an error is thrown.